src/targets/public/main.jsx

```diff
@@ -0,0 +1,73 @@
+/* global cozy */
+import React from 'react'
+import { render } from 'react-dom'
+import { Provider } from 'react-redux'
+import { Router, Route, Redirect, hashHistory } from 'react-router'
+import { I18n } from 'cozy-ui/react/I18n'
+
+import configureStore from '../..//store/configureStore'
+import Layout from '../../components/Layout'
+
+import LightFolderView from '../../components/LightFolderView'
+
+document.addEventListener('DOMContentLoaded', init)
+
+const arrToObj = (obj = {}, [key, val = true]) => {
+  obj[key] = val
+  return obj
+}
+
+const getQueryParameter = () => window
+  .location
+  .search
+  .substring(1)
+  .split('&')
+  .map(varval => varval.split('='))
+  .reduce(arrToObj, {})
+
+function init () {
+  const lang = document.documentElement.getAttribute('lang') || 'en'
+  const root = document.querySelector('[role=application]')
+  const data = root.dataset
+  const { id, sharecode, directdownload } = getQueryParameter()
+
+  if (data.cozyDomain) {
+    cozy.client.init({
+      cozyURL: `//${data.cozyDomain}`,
+      token: sharecode
+    })
+  }
+
+  if (directdownload) {
+    console.log('direct download')
+    cozy.client.files.getDownloadLinkById(id).then(
+      link => window.open(link, '_SELF')
```

Comment:
@y-lohse with the `'_SELF'` argument, it opens in the same exact tab.

- [x] find a workaround for _«it also gets blocked by the popup blocker. Not sure if we can work around this :/»_